### PR TITLE
type selection in proto queries and commands

### PR DESCRIPTION
**Pull request by iceseer**
Signed-off-by: iceseer <iceseer@gmail.com>

### Description of the Change
Performance refactoring.

### CODE REVIEW

**Comment by iceseer** [comment]
Throw in default was discussed in private with Andrey Lebedev. In this code location such a values absolutely unexpected. To avoid situations of adding new value to enum without adding it's we should use tests.

**Comment by MBoldyrev** [comment]
can you please remove this test as obsolete `test/module/shared_model/backend_proto/shared_proto_commands_test.cpp`
and rearrange variant entries in alphabetical order in these files
```
shared_model/backend/protobuf/commands/impl/proto_command.cpp
shared_model/interfaces/commands/command.hpp
shared_model/interfaces/commands/command_variant.hpp
```

File: shared_model/backend/protobuf/queries/impl/proto_query.cpp [hunk 1]
```diff
@@ -150,3 +170,5 @@ namespace shared_model {
 
   }  // namespace proto
 }  // namespace shared_model
+
+#undef IROHA_BIND_TYPE
```
Comment (MBoldyrev) [inline shared_model/backend/protobuf/queries/impl/proto_query.cpp:174]: I think it would be better to `undef` macro right after its last usage.

Comment (iceseer) [reply shared_model/backend/protobuf/queries/impl/proto_query.cpp:174]: It's right there. It's pretty to save the scope consistence.

Comment (iceseer) [reply shared_model/backend/protobuf/queries/impl/proto_query.cpp:174]: Looks better.

**Pull request merged**